araproc/analysis/hilbert.py

```diff
@@ -0,0 +1,97 @@
+# this calculates Hilbert envelope SNR for each channel and their average
+
+import numpy as np
+from scipy.signal import hilbert
+
+from araproc.analysis import snr
+from araproc.framework import waveform_utilities as wfu
+
+
+
+def get_envelope(waveform):
+
+    """
+    Calculates the Hilbert envelope a trace.
+
+    Parameters
+    ----------
+    waveform: TGraph
+        A TGraph of the waveform.
+
+    Returns
+    -------
+    hill: array
+        Hilbert envelope of the waveform.
+    """ 
+
+    _, trace = wfu.tgraph_to_arrays(waveform)
+   
+    hill = np.abs(hilbert(trace))
+    
+    return hill
+
+
+
+def get_hill_snr(waveform):
+
+    """
+    Calculates Hilbert envelope SNR of a single voltage trace.
+
+    Parameters
+    ----------
+    waveform: TGraph
+        A TGraph of the waveform.
+
+    Returns
+    -------
+    hill_snr : float
+        The Hilbert envelope SNR of the waveform.
+    """
+
+    hill = get_envelope(waveform)
+    hill_max_idx = np.argmax(hill)
+    hill_max = hill[hill_max_idx]
+    hill_rms = snr.get_min_segmented_rms(hill)
+
+    if(hill_rms == 0.0):
+      return 0
+
+    hill_snr = hill_max/hill_rms
+
+    return hill_snr
+
+
+
+
+def get_avg_hill_snr(wave_bundle, excluded_channels = []):
+
+    """
+    Calculates channel-wise averaged Hilbert envelope SNR.
+
+    Parameters
+    ----------
+    wave_bundle: dict of TGraphs
+        Dictionary of waveform TGraphs to be averaged.
+    excluded_channels: list
+        List of dictionary keys to exclude from average.
+
+    Returns
+    -------
+    avg_hill_snr : float
+        The average Hilbert envelope SNR.
+    """
+
+    chans = sorted(list(wave_bundle.keys()))
```

Comment:
I'm not sure the sort is necessary here, so could drop it.

Reply:
right, unnecessary!